### PR TITLE
CSECURITY-30: Added the test distribution to and change the OSGi tests to use test distribution.

### PR DESCRIPTION
**Pull request by JKAUSHALYA**
https://wso2.org/jira/browse/CSECURITY-30

### CODE REVIEW

File: tests/distribution/pom.xml [hunk 1]
```diff
@@ -0,0 +1,298 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (omindu) [inline tests/distribution/pom.xml]: Add licence header.

File: tests/osgi-tests/pom.xml [hunk 2]
```diff
@@ -163,35 +163,34 @@
...
-                                    <resources>
-                                        <resource>
-                                            <directory>src/test/resources/carbon-home</directory>
-                                        </resource>
-                                    </resources>
+                                    <artifactItems>
+                                        <artifactItem>
+                                            <groupId>org.wso2.carbon.security.userstore.test</groupId>
+                                            <artifactId>org.wso2.carbon.security.userstore.test.distribution</artifactId>
+                                            <version>1.0.0-SNAPSHOT</version>
```
Comment (omindu) [inline tests/osgi-tests/pom.xml]: Is it necessary to define the version here ?

File: tests/osgi-tests/pom.xml [hunk 3]
```diff
@@ -203,20 +202,51 @@
...
+                                <phase>generate-resources</phase>
+                                <goals>
+                                    <goal>copy-resources</goal>
+                                </goals>
+                                <configuration>
+                                    <outputDirectory>${project.build.directory}/wso2carbon-kernel-${project.version}/conf/</outputDirectory>
+                                    <overwrite>true</overwrite>
+                                    <resources>
+                                        <resource>
+                                            <directory>src/test/resources/conf</directory>
```
Comment (omindu) [inline tests/osgi-tests/pom.xml]: Is it possible to copy individual files rather than copying the entire folder ?

Comment (JKAUSHALYA) [inline tests/osgi-tests/pom.xml]: Entire conf folder is required here. So no use of coping each file individually. And there is underlying folder structure to be kept.

**Pull request merged**